.github/workflows/workflow.yml

```diff
@@ -197,3 +224,62 @@ jobs:
       - run: yarn start ../../rococo-local/config.json & cd ../wait-for-inclusion && npm start 200
         name: Launch rococo local & wait for inclusion
         working-directory: scripts/polkadot-launch
+  
+  build-and-publish-docker-image:
+    needs: [pop-runner-instance, build]
+    runs-on: ${{ needs.pop-runner-instance.outputs.runner }}
+    env:
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      ECR_URL: 601305236792.dkr.ecr.eu-west-1.amazonaws.com
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/download-artifact@v2
+        name: Download basilisk binary
+        with:
+          name: basilisk
+          path: docker
+      - name: Sleep for 60 seconds
+        run: sleep 60
+      - name: install docker
+        run: sudo snap install docker
+      - name: Sleep for 30 seconds
+        run: sleep 30s
+      - name: build docker image
+        run: sudo docker build -t basilisk:latest docker
+      - name: get ecr credentials
+        run: aws ecr get-login-password --region eu-west-1 | sudo docker login --username AWS --password-stdin ${{ env.ECR_URL }}
+      - name: push image to ecr repository
+        run: sudo docker tag basilisk:latest ${{ env.ECR_URL }}/basilisk
+      - run: sudo docker push ${{ env.ECR_URL }}/basilisk:latest
```

Comment:
tag the images based on the branch, `master -> latest`, `pr-branch -> parsed-branchname/commit` hash

Reply:
@lumir-mrkva can you advise the syntax you have in mind?

Reply:
major problem is taht you cannot use `/` in tag so you have to replace it

so for example if the build is from branch `feat/awesome-feature` it would be tagged as `pr-feat-awesome-feature`

later we can also trigger build on git tags and in that case it would tag it without parsing like `v11.0.1` for example

Reply:
Okay, so, our main build would be the one tagged as `master`. For the branches, I will replace `/` with - before naming.

Reply:
master branch should be tagged as `latest`, as that is default tag that docker tries to pull

Reply:
@fakirAyoub this is still an issue